### PR TITLE
chore: document steps for generating changelog

### PR DESCRIPTION
**Pull request by vcrfxia**
### Description 

Documents the steps I've been using to generate the auto-generated changelog for each release. Also checks the processing script into version control. Apologies for the sloppiness.

### Testing done 

Followed the README steps.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: scripts/changelog/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+Suppose you're interested in generating a changelog of all changes in ksqlDB 0.8.0 that are new as of ksqlDB 0.7.0. Refer to these versions as `NEW_VERSION` and `OLD_VERSION`, respectively. The same process applies for other new/old ksqlDB version combinations as well (e.g., 0.8.1 and 0.8.0, or 0.8.0 and 0.7.1).
+
+These steps assume you're in the root of the `ksql` directory.
+
+### List commits on `NEW_VERSION`
+
+Start by listing all commits from the `NEW_VERSION`, which requires checking out the relevant branch/tag/commit. If the release tag exists:
+```
+git fetch --all --tags --prune
+git checkout tags/v0.8.0-ksqldb -b v0.8.0-ksqldb
```
Comment (colinhicks) [inline scripts/changelog/README.md:25]: the branch name should not match the tag name, no?
```suggestion
git checkout tags/v0.8.0-ksqldb -b 0.8.0-ksqldb
```

File: scripts/changelog/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+```
+conventional-changelog -p angular -i new_commits.txt -s -r 0
+```
+
+### List commits on `OLD_VERSION`
+
+Next, repeat the process to list commits from the `OLD_VERSION`. A tag for this release should already exist, so check out the tag:
+```
+git fetch --all --tags --prune
+git checkout tags/v0.7.0-ksqldb -b v0.7.0-ksqldb
```
Comment (spena) [inline scripts/changelog/README.md:48]: This causes ambiguous errors in some environments when running the conventional-changelog. To make it work, I did not create the branch:
```
# git checkout tags/v0.9.0-rc294-ksqldb -b v0.9.0-rc294-ksqldb
Switched to a new branch 'v0.9.0-rc294-ksqldb'
[root@ksql-centos ksql]# conventional-changelog -p angular -i new_commits.txt -s -r 0
Error: Error in git-raw-commits: warning: refname 'v0.9.0-rc294-ksqldb' is ambiguous.
```
```
$ git checkout tags/v0.7.0-ksqldb
Note: switching to 'tags/v0.9.0-rc294-ksqldb'.
...
$ conventional-changelog -p angular -i new_commits.txt -s -r 0
(base) sergio@pop-os:~/Development/ksql$
```

File: scripts/changelog/README.md (hunk 1 above)
Comment (vcrfxia) [reply scripts/changelog/README.md]: It shouldn't matter? @spena said he was having issues with ambiguous references yesterday but using a different branch name didn't resolve the issue.

Regardless, I've updated the PR to use @spena 's suggestion of checking out the reference without creating a branch.

File: scripts/changelog/README.md (hunk 2 above)
Comment (vcrfxia) [reply scripts/changelog/README.md]: I'm still confused where this error is coming from and why it only affects your setup but not mine, but the workaround sounds good so I've updated the doc. Thanks!

File: scripts/changelog/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+Now you're ready to run the script that generates a changelog based on the difference between the two lists of commits from above. In other words, the changelog consists of new commits on `NEW_VERSION` not present on `OLD_VERSION`.
+
+Switch back to master to ensure you have the latest version of the script:
+```
+git checkout master
+```
+and run
+```
+python3 scripts/changelog/process_commit_files.py --release_version 0.8.0 --previous_version 0.7.0 --output_file CHANGELOG_additions.md
```
Comment (spena) [inline scripts/changelog/README.md:65]: I get this error with the script:
```
$ python3 scripts/changelog/process_commit_files.py --release_version 0.8.0 --previous_version 0.7.0 --output_file CHANGELOG_additions.md
Traceback (most recent call last):
  File "scripts/changelog/process_commit_files.py", line 330, in <module>
    all_commits = parse_changelog(args.new_commits_file)
  File "scripts/changelog/process_commit_files.py", line 59, in parse_changelog
    fork_name = _get_fork_name(line)
  File "scripts/changelog/process_commit_files.py", line 124, in _get_fork_name
    start_ind = header.index('https://github.com/')
ValueError: substring not found
```

I noticed the commits file do not include the fork name:
`# [](/compare/v0.8.0-ksqldb...v) (2020-05-08)`

I don't know why the conventional-changelog does not include the https://github.com. The version I have is 2.0.21.

Comment (vcrfxia) [reply scripts/changelog/README.md:65]: What the output of `git remote -v` for you?

Comment (vcrfxia) [reply scripts/changelog/README.md:65]: Also to double-check, which branch/tag were you on when you generated the commit file (using the `conventional-changelog` command)?

Comment (spena) [reply scripts/changelog/README.md:65]:
```
$ git remote -v
sergio  git@github.com:spena/ksql (fetch)
sergio  git@github.com:spena/ksql (push)
upstream        git@github.com:confluentinc/ksql (fetch)
upstream        git@github.com:confluentinc/ksql (push)
```

For the branch/tag question. I tried different options with same results. For instance:
* git checkout tags/v0.8.0-ksqldb
* git checkout tags/v0.8.0-ksqldb -b test

The above two ways allow me to run the conventional-changelog, but the commit files still have the same header. I just tried both.

**Comment by vcrfxia** [comment]
Merging this PR as @stevenpyzhang confirmed he was able to follow the steps out of the box in order to generate the 0.10.0 changelog.

@spena I'm not sure how to debug your setup remotely. Maybe we can hop on a call for some live debugging at some point?

**Pull request merged**